horace_core/sqw/@Experiment/set_instrument.m

```diff
@@ -0,0 +1,86 @@
+function   obj = set_instrument(obj,instr_or_fun,varargin)
+% add or reset instrument, related to the given experiment class
+%
+if isa(instr_or_fun,'IX_inst')
+    if numel(instr_or_fun) == 1
+        obj.instruments.unique_objects = instr_or_fun;
+        if obj.instruments.n_runs ~= obj.n_runs
+            inst = obj.instruments;
+            inst = inst.expand_runs(obj.n_runs); % will work only for single unique instrument
+            obj.instruments = inst;
+        end
+    else
+        inst = obj.instruments;
+        if numel(instr_or_fun) ~= inst.n_runs
+            error('HORACE:Experiment:invalid_argument',...
+                'Multiple instruments (N=%d) provided as input for set_instrument function but the number is not 1 and not equal to the number of runs defined in container (%d)', ...
+                numel(instr_or_fun),inst.n_runs)
+        end
+        n_runs = inst.n_runs;
+        for i=1:n_runs
+            inst(i) = instr_or_fun(i);
+        end
+        obj.instruments = inst;
+    end
+elseif isa(instr_or_fun,'unique_objects_container') && strcmp(instr_or_fun.baseclass,'IX_inst')
+    if instr_or_fun.n_runs ~= obj.n_runs
+        if instr_or_fun.n_runs == 1
+            instr_or_fun = instr_or_fun.expand_runs(obj.n_runs);
+        else
+            error('HORACE:Experiment:invalid_argument',...
+                'Attempt to set instrument as container with %d unique objects, but Experiment contains %d runs so only 1 unique or %d different insruments allowed in the container', ...
+                obj.n_runs,instr_or_fun.n_runs,obj.n_runs)
+        end
+    end
+    obj.instruments = instr_or_fun;
+
+elseif isa(instr_or_fun,"function_handle")
```

Comment:
Why `string` here not `char`?
```suggestion
elseif isa(instr_or_fun,'function_handle')
```

Reply:
both work, in majority of places we use `char` (historically, strings are introduced recently)